<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitXMLReporter -->
<testsuite ignored="0" hostname="EPINHYDW08AD" failures="2" tests="13" name="HomeTaskTest" time="0.281" errors="0" timestamp="2024-07-02T00:58:33 IST">
  <testcase classname="Tests.CalculatorTests" name="addition" time="0.009"/>
  <testcase classname="Tests.CalculatorTests" name="addition_NegativeNumbers1" time="0.009"/>
  <testcase classname="Tests.CalculatorTests" name="addition_NegativeNumbers" time="0.009"/>
  <testcase classname="Tests.CalculatorTests" name="addition" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="addition" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="addition_PositiveAndNegativeNumbers1" time="0.002"/>
  <testcase classname="Tests.CalculatorTests" name="addition_PositiveNumbers" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="division_FindingRemainder" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="multiplication_NegativeNumbers" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="multiplication_NegativeNumbers1" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="multiplication_postiveNumbers" time="0.001"/>
  <testcase classname="Tests.CalculatorTests" name="testDivision_ByZero" time="0.001">
    <failure type="java.lang.NumberFormatException" message="Attempt to divide by zero">
      <![CDATA[java.lang.NumberFormatException: Attempt to divide by zero
at com.epam.tat.module4.Calculator.div(Calculator.java:34)
at Tests.CalculatorTests.testDivision_ByZero(CalculatorTests.java:88)
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:840)
... Removed 13 stack frames]]>
    </failure>
  </testcase> <!-- testDivision_ByZero -->
  <testcase classname="Tests.CalculatorTests" name="testDivideByZero" time="0.001">
    <failure type="org.testng.TestException" message="
Expected exception of type class java.lang.ArithmeticException but got java.lang.NumberFormatException: Attempt to divide by zero">
      <![CDATA[org.testng.TestException: 
Expected exception of type class java.lang.ArithmeticException but got java.lang.NumberFormatException: Attempt to divide by zero
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.NumberFormatException: Attempt to divide by zero
at com.epam.tat.module4.Calculator.div(Calculator.java:34)
at Tests.CalculatorTests.testDivideByZero(CalculatorTests.java:114)
... 12 more
... Removed 17 stack frames]]>
    </failure>
  </testcase> <!-- testDivideByZero -->
</testsuite> <!-- HomeTaskTest -->
